### PR TITLE
Replace C++ enum code generator with CMake script

### PR DESCRIPTION
**Pull request by Neverlord**
We currently have auto-generated files in the repository. This is of course of big no-no, but we ended up with this setup since the files are generated from a C++ tool and the setup crashes and burns when doing cross-compilation.

By using a CMake script to do the code generation instead, we can safely generate the files during the build phase and drop the auto-generated code from the code base once again.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -204,51 +203,24 @@ if(NOT TARGET uninstall)
...
-else()
-  function(caf_add_enum_consistency_check hpp_file cpp_file)
-    # nop
-  endfunction()
-endif()
-
 # -- utility functions ---------------------------------------------------------
 
+# adds a consistency check that verifies that `cpp_file` is still valid by
+# re-generating the file and comparing it to the existing file
```
Comment (josephnoir) [inline CMakeLists.txt:209]: This is wrong, I think. There is not consistency check. This simply generates the files, right?

File: cmake/caf-generate-enum-strings.cmake [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      scan_mode()
+    else()
+      # reached the end
+      if(NOT enum_name)
+        message(FATAL_ERROR "No enum found in input file")
+      endif()
+      if(NOT enum_values)
+        message(FATAL_ERROR "No enum values found for ${enum_name}")
+      endif()
+      write_enum_file()
```
Comment (josephnoir) [inline cmake/caf-generate-enum-strings.cmake:126]: (Why is this not called `generate_mode`? Or why do the `_mode` functions not have similarly speaking names, e.g., `seek_enum` and `scan_cases`.)

Comment (Neverlord) [reply cmake/caf-generate-enum-strings.cmake:126]: That's no longer a processing mode that processes lines. This step is writing the file and then returning from the function/script. But I did gave the two modes some better name.

Comment (josephnoir) [reply cmake/caf-generate-enum-strings.cmake:126]: :)

**Review by josephnoir — APPROVED** [review]

**Pull request merged**